app/src/main/java/com/allan/boardbuddies/activities/CanvasActivity.java

```diff
@@ -0,0 +1,97 @@
+package com.allan.boardbuddies.activities;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.view.ViewTreeObserver;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.EditText;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+
+import com.allan.boardbuddies.R;
+import com.allan.boardbuddies.views.CanvasView;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class CanvasActivity extends AppCompatActivity {
+    private EditText editTextTitle;
+    private EditText editTextContent;
+    private File filePath;
+    private String localFilename;
+    private String localTitle;
+    private String localContent;
+    private int localPosition;
+    private CanvasView paint;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_canvas);
+        Toolbar toolbar = findViewById(R.id.edit_canvas_toolbar);
+        paint = findViewById(R.id.canvas_view);
+        ViewTreeObserver viewTreeObserver = paint.getViewTreeObserver();
+        if (viewTreeObserver.isAlive()) {
+            viewTreeObserver.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+                @Override
+                public void onGlobalLayout() {
+                    paint.getViewTreeObserver().removeOnGlobalLayoutListener(this);
```

Comment:
If I understand correctly, it tells my canvas what is the actual available space to work with after the layout has been finalized/changed (hence the listener)

Again, if I understand correctly, I can't reuse because the global layout will change and the object will not reflect that so it needs to be called from the source